### ISSUE TITLE
Unlock ability to create spool with "higher weight" than theoretically possible

### ISSUE THREAD

**Issue by mdziekon** [Donkie/Spoolman#299]
**Is your feature request related to a problem? Please describe.**

Currently, when adding new spools, Spoolman always assumes that any new spool is "Filament weight" + "Empty spool weight". In theory, this is fine, but as we know it, every spool can vary slightly in weight - sometimes it's a bit less, sometimes it's a bit more. In my case, it's usually the "more" variant, usually 10, 20, sometimes 50 grams more than advertised.

Dealing with "less" works fine - you simply measure your entire spool, and put that weight into the `Measured weight` field...

...but it's impossible to insert spools with "more" weight - even if you do put the exact measured weight, the frontend will show negative number in the `Used weight` field, but more importantly, won't allow you to create / save that spool, as it doesn't pass the backend validation.

**Describe the solution you'd like**

I'd like to be able to create spools with more weight than their theoretical initial weight.

**Describe alternatives you've considered**

Right now, all I can do is put a note for each such spool, and then manually edit spool's data once I use up some filament.

**Comment by MaffooClock** [comment]
Yep, [the API model specifies a constraint](https://github.com/Donkie/Spoolman/blob/a68e0d59bd18983403cc56497bdc9bca84285886/spoolman/api/v1/models.py#L188C32-L188C36) that the value should be greater than or equal to zero, which the web client violates when sending a negative value for `used_weight`.

I suppose a workaround would be to create our Filament records with a weight slightly greater than the actual weight (e.g. 1,010g for a 1Kg spool) to prevent this -- from what I've gathered (and I could be wrong), it doesn't appear that the Filament weight has any purpose other than to constrain the Spool weight.

-----

> [!NOTE]
> Secondary bug discovered

While testing this, I noticed that after receiving the HTTP 422 exception for the failed PATCH request, if you click the Save button again without changing anything, the update succeeds, and that's because the `used_weight` parameter changes back to 0 instead of using the calculated value in the Used Weight field.

I see that a new GET request is sent immediately after the failed PATCH request, the response of which is a JSON object that includes a `used_weight` property.  The Used Weight field in the UI still shows the calculated value, but the `used_weight` value retrieved in the GET request is being used instead.

**Comment by dragoonmc** [comment]
I agree that this would be ideal, Filament manufacturers aren't perfect and might supply a spool that is heavy

**Comment by spyder007** [comment]
I created a discussion that are my thoughts on improving filament/spool management.  If you could see if it makes sense and maybe put some comments in, that would be great!

#338
